ckanext/canada/helpers.py

```diff
@@ -774,6 +776,53 @@ def date_field(field, pkg):
     return pkg.get(field, None)
 
 
+def goc_auth():
+    remote_addr = request.headers.get('X-Forwarded-For') or \
+                  request.environ.get('REMOTE_ADDR')
+    if not __registry_network_access(remote_addr):
+        # can't return 403 because it shows the Login button
+        return t.abort(423, _(u'This application is only available to authorized '
+                            u'Government of Canada departments and agencies. '
+                            u'Please contact the support team at '
+                            u'open-ouvert@tbs-sct.gc.ca to request access.'))
+
+
+def __registry_network_access(remote_addr):
```

Comment:
@RabiaSajjad as the auth function uses this, it would be good to check `flask.has_request_context()` here. As if we are outside the Request context, it is generally tests or system stuff. E.g.
```
from flask import has_request_context
.....
if not has_request_context():
    return True
```